libbeat/monitoring/metrics.go

```diff
@@ -232,3 +234,48 @@ func fullName(r *Registry, name string) string {
 	}
 	return r.name + "." + name
 }
+
+// Timestamp is a timestamp variable satisfying the Var interface.
+type Timestamp struct {
+	mu sync.RWMutex
+	ts time.Time
+}
+
+// NewTimestamp creates and registeres a new timestamp variable.
+func NewTimestamp(r *Registry, name string, opts ...Option) *Timestamp {
+	if r == nil {
+		r = Default
+	}
+
+	v := &Timestamp{}
+	addVar(r, name, opts, v, makeExpvar(func() string {
+		return v.toString()
+
+	}))
+	return v
+}
+
+func (v *Timestamp) Set(t time.Time) {
+	v.mu.Lock()
+	defer v.mu.Unlock()
+
+	v.ts = t
+}
+
+func (v *Timestamp) Get() time.Time {
+	v.mu.RLock()
+	defer v.mu.RUnlock()
+
+	return v.ts
+}
+
+func (v *Timestamp) Visit(_ Mode, vs Visitor) {
+	vs.OnString(v.toString())
+}
+
+func (v *Timestamp) toString() string {
+	if v.ts.IsZero() {
+		return ""
+	}
+	return v.ts.Format(common.TsLayout)
```

Comment:
how about caching the string?

```
if v.cached == "" {
  v.cached = v.ts.Format(common.TsLayout)
}
return v.cached
```

If the timestamp is updated we reset `cached` to `""`.